oauth2/oauth2-client/src/main/java/org/springframework/security/oauth2/client/registration/ClientRegistration.java

```diff
@@ -698,6 +704,21 @@ private void validatePasswordGrantType() {
 			Assert.hasText(this.tokenUri, "tokenUri cannot be empty");
 		}
 
+		private void validateAuthorizationGrantTypes() {
+			List<AuthorizationGrantType> authorizationGrantTypes =
+					Arrays.asList(AuthorizationGrantType.AUTHORIZATION_CODE, AuthorizationGrantType.CLIENT_CREDENTIALS,
+							AuthorizationGrantType.REFRESH_TOKEN, AuthorizationGrantType.IMPLICIT,
+							AuthorizationGrantType.PASSWORD);
```

Comment:
I think it would be good if this was defined as an immutable constant instead of created each time.

Reply:
agreed, wasn't sure if I should put this constant in `ClientRegistration` or in `AuthorizationGrantType`. What do you recommend?